src/main/java/frc/robot/shooter/RealShooterIO.java

```diff
@@ -0,0 +1,92 @@
+package frc.robot.shooter;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.CANSparkBase.IdleMode;
+
+import au.grapplerobotics.ConfigurationFailedException;
+import au.grapplerobotics.LaserCan;
+import au.grapplerobotics.LaserCan.RangingMode;
+import au.grapplerobotics.LaserCan.TimingBudget;
+import frc.robot.Constants;
+
+public class RealShooterIO implements IShooterIO {
+   
+    private CANSparkMax shooterMotorTop;
+    private CANSparkMax shooterMotorBottom;
+    private CANSparkMax indexerMotorTop;
+    private CANSparkMax indexerMotorBottom;
+
+    private RelativeEncoder shooterMotorTopEncoder;
+    private RelativeEncoder shooterMotorBottomEncoder;
+    private RelativeEncoder indexerMotorTopEncoder;
+    private RelativeEncoder indexerMotorBottomEncoder;
+
+    private LaserCan laserCAN;
+
+    public RealShooterIO() {
+        shooterMotorTop = new CANSparkMax(Constants.Shooter.SHOOTER_MOTOR_TOP, CANSparkMax.MotorType.kBrushless);
+        shooterMotorBottom = new CANSparkMax(Constants.Shooter.SHOOTER_MOTOR_BOTTOM, CANSparkMax.MotorType.kBrushless);
+        indexerMotorTop = new CANSparkMax(Constants.Shooter.INDEXER_MOTOR_TOP, CANSparkMax.MotorType.kBrushless);
+        indexerMotorBottom = new CANSparkMax(Constants.Shooter.INDEXER_MOTOR_BOTTOM, CANSparkMax.MotorType.kBrushless);
+
+        indexerMotorTop.setIdleMode(IdleMode.kBrake);
+        indexerMotorBottom.setIdleMode(IdleMode.kBrake);
+
+        shooterMotorTop.setInverted(true);
+        shooterMotorBottom.setInverted(true);
+
+        indexerMotorTop.setInverted(false);
+        indexerMotorBottom.setInverted(false);
+
+        shooterMotorTopEncoder = shooterMotorTop.getEncoder();
+        shooterMotorBottomEncoder = shooterMotorBottom.getEncoder();
+        indexerMotorTopEncoder = indexerMotorTop.getEncoder();
+        indexerMotorBottomEncoder = indexerMotorBottom.getEncoder();
+
+        laserCAN = new LaserCan(Constants.Shooter.LASER_CAN);
+        
+        // configure the laserCAN sensor
+        try {
+            laserCAN.setRangingMode(RangingMode.SHORT);
+            laserCAN.setTimingBudget(TimingBudget.TIMING_BUDGET_20MS);
+        } catch (ConfigurationFailedException e) {
+        	e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void setIndexerOutput(double power) {
+        indexerMotorTop.setVoltage(Math.min(1, Math.max(power, -1)) * 12);
+        indexerMotorBottom.setVoltage(Math.min(1, Math.max(power, -1)) * 12);
```

Comment:
You should take this math stuff and move it to a helper method

Reply:
Part of me wants to say we shouldn't do it at all and should alctually throw an exception if a value outside of -1 to 1 comes in here. We would want to know